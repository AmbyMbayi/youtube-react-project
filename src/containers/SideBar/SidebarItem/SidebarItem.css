.sidebar__item {
    color: white;
    padding-top: 15px;
    padding-bottom: 15px;
    display: flex;
    align-items: center;
}

.sidebar__item:hover {
    background-color: #026e6e;
    cursor: pointer;
}

.sidebar__item label {
    margin-left: 20px;
    font-size: 18x;
}

.sidebar__item i {
    font-size: 25px;
    margin-left: 15px;
}