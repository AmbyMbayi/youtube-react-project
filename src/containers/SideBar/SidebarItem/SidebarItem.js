import React from 'react';
import './SidebarItem.css';

function SidebarItem(props) {
	return (
		<div className="sidebar__item">
			<i className={props.style} />
			<label>{props.label}</label>
		</div>
	);
}

export default SidebarItem;
