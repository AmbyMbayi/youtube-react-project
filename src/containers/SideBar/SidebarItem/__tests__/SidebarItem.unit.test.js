import React from 'react';
import { shallow } from 'enzyme';
import SidebarItem from '../SidebarItem';

describe('SidebarItem', () => {
	test('renders empty SidebarItem', () => {
		const wrapper = shallow(<SidebarItem />);
		expect(wrapper).toMatchSnapshot();
	});
	test('renders SidebarItem taht navigates to /feed/trending', () => {
		const wrapper = shallow(<SidebarItem label="trending" />);
		expect(wrapper).toMatchSnapshot();
	});
});
