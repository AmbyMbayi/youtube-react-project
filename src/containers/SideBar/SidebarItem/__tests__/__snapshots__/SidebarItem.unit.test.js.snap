// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarItem renders SidebarItem taht navigates to /feed/trending 1`] = `
<div
  className="sidebar__item"
>
  <i />
  <label>
    trending
  </label>
</div>
`;

exports[`SidebarItem renders empty SidebarItem 1`] = `
<div
  className="sidebar__item"
>
  <i />
  <label />
</div>
`;
